app/src/main/kotlin/com/simplemobiletools/keyboard/helpers/ShiftState.kt

```diff
@@ -0,0 +1,82 @@
+package com.simplemobiletools.keyboard.helpers
+
+import android.content.Context
+import android.text.InputType
+import com.simplemobiletools.keyboard.extensions.config
+
+enum class ShiftState {
+    OFF,
+    ON_ONE_CHAR,
+    ON_PERMANENT;
+
+    companion object {
+        private val endOfSentenceChars: List<Char> = listOf('.', '?', '!')
+        private const val MIN_TEXT_LENGTH = 2
+
+        /**
+         * Input Type exceptions for capitalizing.
+         */
+        private val inputTypeExceptions = listOf(
+            InputType.TYPE_TEXT_VARIATION_PASSWORD,
+            InputType.TYPE_TEXT_VARIATION_VISIBLE_PASSWORD,
+            InputType.TYPE_TEXT_VARIATION_WEB_PASSWORD,
+            InputType.TYPE_NUMBER_VARIATION_PASSWORD,
+            InputType.TYPE_TEXT_VARIATION_EMAIL_ADDRESS,
+            InputType.TYPE_TEXT_VARIATION_WEB_EMAIL_ADDRESS,
+            InputType.TYPE_TEXT_VARIATION_URI
+        )
+
+        fun getDefaultShiftState(context: Context, inputTypeClassVariation: Int): ShiftState {
+            if (isInputTypeAllowedCapitalizing(inputTypeClassVariation)) {
+                return OFF
+            }
+            return when (context.config.enableSentencesCapitalization) {
+                true -> ON_ONE_CHAR
+                else -> OFF
+            }
+        }
+
+        fun getShiftStateForText(context: Context, inputTypeClassVariation: Int, text: String?): ShiftState {
+            if (isInputTypeAllowedCapitalizing(inputTypeClassVariation)) {
+                return OFF
+            }
+            return when {
```

Comment:
this is a basic boolean check too, avoid complicating code without any reason please